### PR TITLE
Feature/repux 663/amazon integration

### PR DESCRIPTION
**Pull request by przemczan**

### CODE REVIEW

File: src/Service/AmazonMWS/AmazonChannelProcessImportOrdersService.php [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+
+                do {
+                    $this->processOrders();
+                } while (!empty($this->nextToken));
+                $repeatRequest = false;
+            } catch (AmazonOrdersException $ex) {
+                if ($ex->getStatusCode() == 503) {
+                    $repeatRequest = true;
+                    $this->writeln(sprintf('Request throttled by amazon'));
+                    sleep(self::THROTTLING_SLEEP_SECONDS); // sorry i'm bad bad bad programmer :(
```
Comment (kduralek) [inline src/Service/AmazonMWS/AmazonChannelProcessImportOrdersService.php]: remove this comment before it goes public

File: src/Service/AmazonMWS/AmazonThrottlingManager.php [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+     * @param $requestId
+     * @param int $quota
+     *
+     * @return int
+     */
+    public function getThrottlingQueueSize($requestId, $quota = 100)
+    {
+        $key = $this->formKey([$requestId, '*']);
+
+// Commented out because of error: "snc_redis.ERROR: Command "SCAN  amazon_throttling:ListOrderItems[...] 2" failed (ERR syntax error)"
```
Comment (kduralek) [inline src/Service/AmazonMWS/AmazonThrottlingManager.php]: remove this comment

File: src/Service/AmazonMWS/AmazonThrottlingManager.php [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+//
+//        $iterator = null;
+//        return count($this->redis->scan($iterator, $key, $quota));
+
+        return count($this->redis->keys($key)); // TODO: Don't use redis `keys` command
```
Comment (kduralek) [inline src/Service/AmazonMWS/AmazonThrottlingManager.php]: same here

File: src/Service/AmazonMWS/CheckAmazonCredentialsService.php [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+    private $encryptionService;
+
+    private $container;
+
+    private $logger;
+
+    private $amazonUrlService;
+
+    private $supportMarketplaseIds = [
```
Comment (kduralek) [inline src/Service/AmazonMWS/CheckAmazonCredentialsService.php]: marketplaCe :) Please move these ids to a configuration file

**Review by kduralek — APPROVED** [review]

File: src/Service/AmazonMWS/AmazonChannelProcessImportOrdersService.php (hunk 1 above)
Comment (przemczan) [reply src/Service/AmazonMWS/AmazonChannelProcessImportOrdersService.php]: I think it already is public :D just not in develop

**Review by gbar-repux — APPROVED** [review]

**Pull request merged**